static/js/app.ts

```diff
@@ -185,6 +185,21 @@ function clearErrors(editor: AceAjax.Editor) {
   }
 }
 
+export function newly_learned_subjects(level: any, code: string) {
+  try {
+    var necessary_commands = ErrorMessages['New_subjects'][level];
+    for ( var i = 0; i < necessary_commands.length; i++ ) {
```

Comment:
ok I know it is a bit nitpicky but... if we just match strings it is not very precise. For example if I write `print random` it would count as a use of random. I think it should be possible to extract a list of commands in `hedy.py` and use that to match here?

(but we can also leave that for your BSc thesis of course and merge this now as it is :D )

Reply:
Wow, we didn't think of that, but `print random` is very clever haha. @Jasper1408 and I haven't found the solution for this yet. Since the official due date for our project is today, I'd like to take the opportunity to finish this for my BSc thesis! 

We pushed solutions to the first 2 comments you posted. We'd like to merge this now as it is.

Reply:
Hahaha yeah kids come up with the cleverest programs :D Totally fine if you pick this up for your thesis, will you make an issue for it an assign yourself?